### PR TITLE
Fix build error due to const qualifiers not matching

### PR DESCRIPTION
**Pull request by amzn-sj**
Signed-off-by: amzn-sj <srikkant@amazon.com>

### CODE REVIEW

File: Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp [hunk 1]
```diff
@@ -253,7 +253,7 @@ AZStd::vector<AZ::Vector3> TerrainSystem::GenerateInputPositionsFromRegion(
 }
 
 AZStd::vector<AZ::Vector3> TerrainSystem::GenerateInputPositionsFromListOfVector2(
-    const AZStd::span<AZ::Vector2>& inPositionsVec2) const
+    const AZStd::span<const AZ::Vector2> inPositionsVec2) const
```
Comment (spham-amzn) [inline Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp:256]: no longer passed by reference?

**Review by mbalfour-amzn — APPROVED** [review]

File: Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp (hunk 1 above)
Comment (amzn-sj) [reply Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp:256]: Nope. It's a span. So it's just passing a range of iterators. This brings it in line with all other span parameters in this file.

Comment (mbalfour-amzn) [reply Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp:256]: @spham-amzn There's no compelling reason to pass a span by reference, it doesn't own data, it just points at data.

**Review by AMZN-ScottR — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by lsemp3d — APPROVED** [review]

File: Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp (hunk 1 above)
Comment (spham-amzn) [reply Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp:256]: then there shouldnt be a need for the const anymore
```suggestion
    AZStd::span<const AZ::Vector2> inPositionsVec2) const
```

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**